### DISCUSSION TITLE
Change branch name in a forked directory

### DISCUSSION THREAD

**Discussion by priyanka-surana** [community/community#22531] [New to GitHub]
<p>We forked the (nf-core/modules) directory into our account (sanger-tol). The main branch for nf-core/modules is named “master”, and so it is the same in sanger-tol/modules. We would like to change the branch name from “master” to “main”. Can we do that without affecting our ability to <code>prefetch upstream</code>? Thank you :slight_smile:</p>

**Comment by queenofcorgis** [comment, marked as answer]
<p>Hi there <a class="mention" href="/priyanka-surana">@priyanka-surana</a>! Just so I understand what you’re doing are you <a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork">syncing a fork from the upsteam repository</a>?.</p>
<p>If so, changing the name may cause you some trouble down the line so if you are going to be doing this often. If you’re currently using the Web UI to sync changes from the upstream repository then it may no longer work after changing the branch name in your repository.</p>
<p>You should still be able to do this <a href="https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/syncing-a-fork#syncing-a-fork-from-the-command-line">via the command line</a>, you you’ll need to explicitly specify the <code>master</code> branch each time you want to merge changes from upstream: <code>git merge upstream/master</code>.</p>

**Comment by priyanka-surana** [comment]
<p>Hi <a class="mention" href="/liligalante">@liligalante</a> Yes my repo <code>sanger-tol/modules</code> is a fork of <code>nf-core/modules</code>. I wanted to rename our head branch from <code>master</code> to <code>main</code> but was worried about how it would affect <code>fetch upstream</code>. From your message it seems that if I change the name, I can still continue getting upstream changes but it has to be via command line and cannot be done via Web UI.</p>
<p>Thank you for the response.</p>

**Comment by queenofcorgis** [comment]
<p>You are correct - do so via the command line :slight_smile: Glad I could be of assistance!</p>
